src/statespace/dart/SO2Joint.cpp

```diff
@@ -5,8 +5,7 @@ namespace statespace {
 namespace dart {
 
 //=============================================================================
-SO2Joint::SO2Joint(
-      ::dart::dynamics::SingleDofJoint* _joint)
+SO2Joint::SO2Joint(::dart::dynamics::GenericJoint<::dart::math::R1Space> *_joint)
```

Comment:
It's intentional for now. DART currently doesn't distinct R and SO(2) using different classes anyways. Instead, we have used a member function that says if the joint configuration space is SO(2) by checking the position limits. I'm thinking of introducing real SO(2) configuration space in the future, but not in DART 6.1.